README.md
![](https://tompidev.com/bl-content/uploads/pages/0bee045ef26a63902c2844b8a5c0c1d3/screenshot.png)

### What is Fontawesome Helper plugin?

FontAwesome Helper is a Bludit plugin which can help for developers quickly review and insert icons into the HTML source.
On click an icon you get the HTML formatted code of the choosen icon like this:

```HTML
<i class="fa fa-adjust"></i>
```

### What you get with this plugin?

* more than 650 icons by the core lineawesome.css file of Bludit
* clickable icons in a Modal window
* a "one click copy" feature which will help you to copy the HTML formatted icon code to the clipboard
* search box to find an icon
* icon size selector

### Languages available

Fontawesome Helper contains the following languages in this release:

* English
* Hungarian
* French (Thanks to [nicobubulle (github.com)](https://github.com/nicobubulle))

### Installing and activating the plugin

* download the package 
* unzip into a folder located on your local PC
* upload the **fontawesome-helper** folder into **bl-plugins** folder on your website

**After successfully upload...**

To using the plugin you have to activate it on the **Settings / Plugins** page on admin area of your website.

### Demo

To accessing the demo admin interface, here is your login data:

**User**: DemoUser 
**Password**: Demo123

[To visit demo site click here...](http://demo.tompidev.com/admin)

### Buy me a coffee...

If you find my scripts helpful please support my work and Buy Me a Coffe. 

All small support can help me and give me power and insiration to create more useful scripts. 

### [Buy Me a Coffee...](https://www.buymeacoffee.com/tompidev)

**Thank you very much!**


changelog.md
## Fontawesome Helper Plugin Changelog

#### v1.0.7 (09-7-2021)

##### Improvements:

* Building the plugin update system

##### Fixes

* updated search plugin (freezing issue fix)
* bug fixes

##### Design

* Release notes modal window corrections
* some other design corrections

#### v1.0.6 (29-3-2021)

* Correct footer URLs

#### v1.0.5 (20-3-2021)

* Some bug fixes
* New language: French (Thanks to [nicobubulle (github.com)](https://github.com/nicobubulle))
* Updated Hungarian language file

#### v1.0.4 (19-9-2020)

##### Improvements:

* search box to fing icons
* new UI design
* icon size selector

#### v1.0.3 (08-9-2020)

* Reordering files in the package

#### v1.0.2 (07-9-2020)

* Initial stable public release


fontawesome-helper/fahelper.php
<?php
if ( (! empty($_SERVER['REQUEST_SCHEME']) && $_SERVER['REQUEST_SCHEME'] == 'https') ||
     (! empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') ||
     (! empty($_SERVER['SERVER_PORT']) && $_SERVER['SERVER_PORT'] == '443') ) {
    $server_request_scheme = 'https';
} else {
    $server_request_scheme = 'http';
}
$pluginDir = $server_request_scheme . '://' . $_SERVER['HTTP_HOST'] . dirname($_SERVER['PHP_SELF']);

$clipboardJs = $pluginDir . '/clipboard.min.js';
$facss = $pluginDir . '/css/line-awesome/css/line-awesome-font-awesome.min.css';

echo "<script src='$clipboardJs'></script>";

$fontawesome = file_get_contents($facss);
preg_match_all('/"}.(.+?):before/', $fontawesome, $class); // icon name e.g. 'fa fa-adjust'
preg_match_all('/:"\\\(.+?)"}/', $fontawesome, $contents); // css ID excluded backslash '\'

$cont = array();
foreach ($contents[1] as $content) {
	$cont[] = $content; // getting css 'content' id of all icons
}
?>

			<?php
			$id = 0;
			foreach ($class[1] as $fa) {
				$id = ++$id;
				$iconName = substr($fa, 3); // getting the icon name without the 'fa-' prefix
				echo "<i id='$cont[$id]' class='fa-icon fa-helper-icon m-1 p-lg-2 text-center border rounded-sm fa fa-3x $fa' data-name='fa-helper-icon' data-icon='$fa' data-iconname='$iconName'><div class='fa-helper-title'>$fa</div></i>";
			}
			?>

<script>
	$(document).ready(function() {

		/*
		 * Clipoard.js - Calling html element
		 */
		new ClipboardJS("#codeCopy");

		/*
		 * Create html formatted code from icon name and put into the code block when user clicks any icon on the list
		 */
		$('i[data-name="fa-helper-icon"]').click(function(e) {
			var title = $(this).attr('data-icon');
			var size = $('input[name="faIconSize"]:checked').val();
			if (size > 1) {
				var faClass = "fa-" + size + "x " + title;
				var faCode = "&lt;i class=\"fa fa-" + size + "x " + title + "\"&gt;&lt;/i&gt;";
			} else {
				var faClass = "fa fa-5x " + title;
				var faCode = "&lt;i class=\"fa " + title + "\"&gt;&lt;/i&gt;";
			}
			// $("#faSample i").switchClass(faClass);
			$("#code").html(faCode);
			$(".code-copy").removeClass("d-none");
		});

		/*
		 * Setting icon size and append into the html formatted code above
		 */
		$('input[name="faIconSize"]').click(function() {
			var size = this.value;
			var code = $("#code").html();
			var regex = /fa\sfa-+?.x/g;
			if (regex.test(code)) {
				if (size > 1) {
					var faSize = code.replace(regex, "fa fa-" + size + "x");
				} else {
					var faSize = code.replace(regex, "fa");
				}
			} else if (size > 1) {
				var regex = /fa\s/g;
				var faSize = code.replace(regex, "fa fa-" + size + "x ");
			}
			$("#code").html(faSize);
		});

		/*
		 * Empty code block, hide copy button and set icon size radio buttons to unchecked on click Reset button
		 */
		$('#resetBtn').click(function() {
			$("#code").html("");
			$(".code-copy").addClass("d-none");
			$('input[name="faIconSize"]').prop("checked", false);
			$('#faIconSize1').prop("checked", true);
			// $("#faIconSearch").val("");
		});

		/*
		 * Empty modal's 'inputs' and hide copy button again when modal window closed
		 */
		$("#faModal").on("hidden.bs.modal", function() {
			$("#code").html("");
			$("#faIconSearch").val("");
			$(".code-copy").addClass("d-none");
			$('input[name="faIconSize"]').prop("checked", false);
			$('#faIconSize1').prop("checked", true);
		});

		/*
		 * Show alert when code copied to clipboard
		 */
		$('#faModal').on('shown.bs.modal', function() {
			$(".code-copy").click(function() {
				$(".notify").toggleClass("notifyactive");
				$("#notifyType").toggleClass("success");

				setTimeout(function() {
					$(".notify").removeClass("notifyactive");
					$("#notifyType").removeClass("success");
				}, 2000);
			});
		});

		// search icons
		// get input field and add 'keyup' event listener
		let searchInput = document.querySelector('#faIconSearch');
		// let searchInput = document.querySelector('#searchFaIcon');
		searchInput.addEventListener('keyup', search);

		// get all elements with data-iconname attribute
		let data = document.querySelectorAll('[data-iconname]');
		let searchTerm = '';
		let name = '';

		function search(e) {
			// get input fieled value and change it to lower case
			searchTerm = e.target.value.toLowerCase();

			data.forEach((elem) => {
				// navigate to p in the title, get its value and change it to lower case
				name = elem.textContent.toLowerCase();
				// if search term not in the title's title hide the title. otherwise, show it.
				name.includes(searchTerm) ? elem.style.display = '' : elem.style.display = 'none';
			});
		}

	});
</script>

fontawesome-helper/plugin.php
<?php

/**
 *  @package        :  FontAwesome Helper
 *  @author         :  Tompidev
 *  @website        :  https://github.com/tompidev
 *  @email          :  support@tompidev.com
 *  @license        :  MIT
 *
 *  @last-modified  :  2021-07-09 20:08:55 CET
 *  @release        :  1.0.7
 **/

class pluginFontAwesomeHelper extends Plugin
{

	/*
	* Create new global variable for plugin 'codeName'
	*/
	public $pluginName = "faPlugin";

	public function init()
	{
		$this->dbFields = array(
			'faOnSidebar' => true
		);
	}

	public function form()
	{
		global $L;
		global $site;

		$html  = '<div class="alert alert-primary" role="alert">';
		$html .= $this->description();
		$html .= '</div>';

		/*
		* Show warning message about new plugin release
		*/
		$html .= '<div class="row d-none" id="'. $this->pluginName .'VersionAlert">' . PHP_EOL;
		$html .= '<div class="col">' . PHP_EOL;
		$html .= '<div class="alert alert-warning alert-dismissible border" role="alert">' . $L->g('new-release-warning') . '' . PHP_EOL;
		$html .= '<a id="learnMore" type="button" class="btn btn-success btn-sm text-light ml-2" data-toggle="modal" data-target="#'. $this->pluginName .'ReleaseNotesModal">' . $L->g('Learn more') . '</a>' . PHP_EOL;
		$html .= '<button type="button" class="close" data-dismiss="alert" aria-label="Close">' . PHP_EOL;
		$html .= '<span aria-hidden="true">&times;</span>' . PHP_EOL;
		$html .= '</button>' . PHP_EOL;
		$html .= '</div>' . PHP_EOL;
		$html .= '</div>' . PHP_EOL;
		$html .= '</div>' . PHP_EOL;

		/*
		* Select to show or hide menuitem on sidebar
		*/
		$html .= PHP_EOL . '<div>';
		$html .= '<label for="faOnSidebar" style="font-size:1.25rem">' . $L->g('Show or hide on Sidebar') . '</label>' . PHP_EOL;
		$html .= '<select id="faOnSidebar" name="faOnSidebar">' . PHP_EOL;
		$html .= '<option value="true" ' . ($this->getValue('faOnSidebar') === true ? 'selected' : '') . '>' . $L->g('Enabled') . '</option>' . PHP_EOL;
		$html .= '<option value="false" ' . ($this->getValue('faOnSidebar') === false ? 'selected' : '') . '>' . $L->g('Disabled') . '</option>' . PHP_EOL;
		$html .= '</select>' . PHP_EOL;
		$html .= '<small class="text-muted">' . $L->g('Show or hide FA helper menuitem on Sidebar') . '</small>' . PHP_EOL;
		$html .= '</div>' . PHP_EOL;

		/*
		* Calling modal "Showroom"
		*/
		$html .= '<a id="loadFaModal" class="btn btn-info btn-lg mt-4 text-white" data-toggle="modal" data-target="#faModal" role="button">' . $L->get('Open showroom') . '</a>';

			/*
		* App footer for plugin version and developer's urls
		*/
		$html .= PHP_EOL . '<div class="text-center pt-3 mt-4 border-top text-muted">' . PHP_EOL;
		$html .= $this->name() . ' - v' . $this->version() . ' @ ' . date('Y') . ' by ' .  $this->author() . PHP_EOL;
		$html .= '</div>' . PHP_EOL;
		$html .= '<div class="text-center">' . PHP_EOL;
		$html .= '<a class="fa fa-2x fa-globe" href="https://www.tompidev.com/" target="_blank" title="Visit TompiDev\'s Website"></a>' . PHP_EOL;
		$html .= '<a class="fa fa-2x fa-github" href="https://www.github.com/tompidev" target="_blank" title="Visit TompiDev on Github"></a>' . PHP_EOL;
		$html .= '<a class="fa fa-2x fa-twitter" href="https://www.twitter.com/tompidev" target="_blank" title="Visit TompiDev on Twitter"></a>' . PHP_EOL;
		$html .= '<a class="fa fa-2x fa-envelope" href="mailto:support@tompidev.com/?subject=Question%20about%20' . $this->name() . '" title="Send me an email"></a>' . PHP_EOL;
		$html .= '<a class="fa fa-2x fa-cubes" href="https://www.tompidev.com/fontawesome-helper" target="_blank" title="Plugin\'s website on tompidev.com"></a>' . PHP_EOL;
		$html .= '</div>' . PHP_EOL;

		/*
		* Modal for Release notes
		*/
		$html .= '
<div class="modal fade" id="'. $this->pluginName .'ReleaseNotesModal" tabindex="-1" aria-labelledby="'. $this->pluginName .'ReleaseNotesModal" aria-hidden="true">
<div class="modal-dialog modal-dialog-centered modal-dialog-scrollable">
<div class="modal-content">
 <div class="modal-header text-center">
	 <h4 class="modal-title w-100">' . $L->g('Release Notes') . '</h4>
 </div>
 <div class="modal-body">
	 <div calss="container">
				<div class="row">
						<div class="col-5 font-weight-bold pr-1 mr-2">' . $L->g('Package') . ':</div>
						<div id="'. $this->pluginName .'PackageName"></div>
				</div>
				<div class="row">
						<div class="col-5 font-weight-bold pr-1 mr-2">' . $L->g('Current version') . ':</div>
						<div id="'. $this->pluginName .'CurrentVersion"></div>
				</div>
				<div class="row">
						<div class="col-5 font-weight-bold pr-1 mr-2">' . $L->g('New version') . ':</div>
						<div id="'. $this->pluginName .'NewVersion"></div>
				</div>
				<div class="row">
						<div class="col-5 font-weight-bold pr-1 mr-2">' . $L->g('Release date') . ':</div>
						<div id="'. $this->pluginName .'ReleaseDate"></div>
				</div>
	 </div>
	 <h5 class="my-3 pt-3 border-top">' . $L->g('What\'s new in this release') . '</h5>
	 <div class="pl-3" id="'. $this->pluginName .'ReleaseNotes"></div>
	 <div id="usufelLinks" class="mt-3 pt-3 border-top">
	 <h5>' . $L->g("Useful Links") . '</h5>
			 <a href="http://demo.tompidev.com/" target="_blank">Demo admin<span class="fa fa-external-link ml-2"></span></a>(' . $L->g('Try Tompidev\'s plugins and themes for FREE') . ')<br>
			 <a href="https://tompidev.com/" target="_blank">Developer\'s website<span class="fa fa-external-link ml-2"></span></a>(' . $L->g('All plugins and themes from Tompidev') . ')
	 </div>
 </div>
 <div class="modal-footer justify-content-center">
		<a id="downloadLink" class="btn btn-primary" href="" target="_blank"><i class="fa fa-download"></i>' . $L->g('Download release') . '</a>
		<a id="changelogLink" class="btn btn-primary" href="" target="_blank"><i class="fa fa-info-circle"></i>' . $L->g('Changelog') . '</a>
		<a id="github" class="btn btn-primary" href="" target="_blank"><i class="fa fa-github"></i>Github</a>
 </div>
</div>
</div>
</div> ' . PHP_EOL;

		/*
		* Modal for Fontawesome icons
		*/
		$html .= '
		<div class="modal fade fa-modal-xl" id="faModal" tabindex="-1" role="dialog" aria-labelledby="faModalLabel" aria-hidden="true">
		<div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
			<div class="modal-content">
				<div class="modal-header justify-content-md-center">
					<div class="notify"><span id="notifyType" class=""></span></div> <!-- Container for alert when code copied to clipboard-->
					<div class="container">
						<div class="input-group mb-3 border rounded bg-light">
							<pre>
								<span class="code-copy border-right d-none" role="button"><i id="codeCopy" class="fa fa-paste btn pb-0" aria-hidden="true" title="' . $L->get('Copy code to clipboard') . '" data-clipboard-target="#code"></i></span>
								<code id="code" class="language-html"></code>
							</pre>
							<div class="input-group-append reset-button">
								<button id="resetBtn" type="button" class="btn btn-primary text-uppercase" data-dismiss="code" aria-label="Reset">' . $L->get('Clean') . '
								</button>
							</div>
						</div>
						<div class="input-group td-input-group">
							<div class="input-group-prepend w-auto">
								<div class="input-group-text td-input-group-text">
									<input class="mr-1" type="radio" name="faIconSize" id="faIconSize2" value="2">
										<span class="form-check-label mt-0 mr-2" for="faIconSize2">2x</span>
									<input class="mr-1" type="radio" name="faIconSize" id="faIconSize3" value="3">
										<span class="form-check-label mt-0 mr-2" for="faIconSize3">3x</span>
									<input class="mr-1" type="radio" name="faIconSize" id="faIconSize4" value="4">
										<span class="form-check-label mt-0 mr-2" for="faIconSize4">4x</span>
									<input class="mr-1" type="radio" name="faIconSize" id="faIconSize5" value="5">
										<span class="form-check-label mt-0 mr-2" for="faIconSize5">5x</span>
									<input class="mr-1" type="radio" name="faIconSize" id="faIconSize1" value="1" checked>
										<span class="form-check-label mt-0" for="faIconSize1">' . $L->get('Default') . ' (1x)</span>
								</div>
							</div>
							<input id="faIconSearch" type="text" class="form-control faIconSearch" placeholder="' . $L->get('Type to search icons') . '">
						</div>
					</div>
				</div>
				<div class="modal-body pt-0 px-0">
					<div id="faContent" class="container mt-3"></div>
				</div>
			</div>
		</div>
	</div>
			' . PHP_EOL;

		return $html;
	}

	public function adminHead()
	{
		$html = '<link rel="stylesheet" type="text/css" href="' . HTML_PATH_PLUGINS . 'fontawesome-helper/css/style.css">' . PHP_EOL;

		return $html;
	}

	public function adminSidebar()
	{
		global $L;

		$html = '';
		if ($this->getValue('faOnSidebar') && $this->getValue('faOnSidebar') == true) {
			$html = '<a id="faNavitem"  class="nav-link" href="' . HTML_PATH_ADMIN_ROOT . 'configure-plugin/' . $this->className() . '" title="' . $L->get('Open Settings page of '.$this->name(). ' plugin') . '">' . $this->name() . '<i id="'. $this->pluginName .'VersionMenuAlert" title="' . $L->get('new-release-warning') . '"></i></a>';
		}

		return $html;
	}

	public function adminBodyEnd()
	{
		global $L;

		$scripts = '<script>' . PHP_EOL;

		/*
		* Loading file into modal
		*/
		$file = DOMAIN_PLUGINS . "fontawesome-helper/fahelper.php";

		$scripts .= '$("#loadFaModal").click(function(){' . PHP_EOL;
		$scripts .= '$("#faContent").load("' . $file . '")' . PHP_EOL;
		$scripts .= '})' . PHP_EOL;

		$scripts .= '</script>' . PHP_EOL;

		/*
		* Version check script
		*
		* Reading data from remote json file
		*/
		$scripts .= '<script>

		function check'.$this->pluginName. 'Version() {

				// console.log("[INFO] [PLUGIN VERSION] Getting list of versions of '.$this->name(). ' plugin.");

				$.ajax({
						url: "https://tompidev.com/release-info/json/'.$this->className(). '.json",
						method: "GET",
						dataType: "json",
						success: function(json) {
								if (json.'. $this->pluginName .'.newVersion > "'.$this->version().'") {
										console.log("[INFO] [PLUGIN VERSION] New '.$this->name(). ' plugin version is available: v" + json.'. $this->pluginName .'.newVersion);
										$("#'. $this->pluginName .'PackageName").html(json.'. $this->pluginName .'.package);
										$("#'. $this->pluginName .'CurrentVersion").html("'.$this->version().'");
										$("#'. $this->pluginName .'NewVersion").html( json.'. $this->pluginName .'.newVersion );
										$("#'. $this->pluginName .'ReleaseDate").html( json.'. $this->pluginName .'.releaseDate );
										var changelogObj, i, j, x = "";
										changelogObj = [ json.'. $this->pluginName .'.changelog ];
										console.log(changelogObj);
										for (i in json.'. $this->pluginName .'.changelog) {
												x += "<h5 class=\"mt-2\">" + json.'. $this->pluginName .'.changelog[i].action + "</h5>";
												for (j in json.'. $this->pluginName .'.changelog[i].items) {
												x += "<span class=\"fa fa-arrow-right ml-2\"></span>" + json.'. $this->pluginName .'.changelog[i].items[j] + "<br>";
												}
										}
										$("#'. $this->pluginName .'ReleaseNotes").html( x );
										$("#'. $this->pluginName .'VersionAlert").removeClass("d-none");
										$("#'. $this->pluginName .'VersionMenuAlert").addClass("fa fa-bell mr-1 text-warning float-right");
										$("#downloadLink").attr("href",  json.'. $this->pluginName .'.downloadLink );
										$("#changelogLink").attr("href", json.'. $this->pluginName .'.changelogLink );
										$("#github").attr("href", json.'. $this->pluginName .'.github );
								}
						},
						error: function(json) {
								console.log("[WARN] [PLUGIN VERSION] There is some issue to get the version status of '.$this->name(). ' plugin.");
						}
				});
		}
		check'.$this->pluginName. 'Version();
		</script>' . PHP_EOL;
		return $scripts;
	}
}
